instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java

```diff
@@ -32,10 +36,18 @@ public static void onEnter(
 
     Context attachedContext = tracer().getServerContext(httpServletRequest);
     if (attachedContext != null) {
-      if (tracer().needsRescoping(attachedContext)) {
+      if (Servlet3HttpServerTracer.needsRescoping(attachedContext)) {
         scope = attachedContext.makeCurrent();
       }
 
+      // We're interested only in the very first suggested name, as this is where the initial
+      // request arrived. There are potential forward and other scenarios, where servlet path
+      // may change, but we don't want this to be reflected in the span name.
+      if (!AppServerBridge.isBetterNameSuggested(attachedContext)) {
+        tracer().updateServerSpanName(httpServletRequest);
+        AppServerBridge.setBetterNameSuggested(attachedContext, true);
+      }
```

Comment:
the SERVER span name is also updated in several of the web framework instrumentations

Reply:
Web Framework instrumentations are expecting the SERVER span to be present and to update the server span name unconditionally. Servlet integration was either _creating_ a new SERVER span or exiting as soon as possible if a SERVER span was already present. This flag is actually needed for the Servlet instrumentation only to indicate that the servlet instrumentation itself has set the name. I probably should rename methods to better reflect what's happening.